tests/specs/lemmas.k

```diff
@@ -58,6 +65,14 @@ module LEMMAS
     rule 0 <=Int keccak( _ )             => true [simplification]
     rule         keccak( _ ) <Int pow256 => true [simplification]
 
+    rule #lookup( M:Map [ keccak(_) <- _ ], I:Int ) => #lookup( M , I )
+      requires 0 <=Int I andBool I <Int 10
+      [simplification]
+
+    rule #lookup( M:Map [ keccak(_) +Int I1:Int <- _ ], I2:Int ) => #lookup( M , I2 )
+      requires 0 <=Int I1 andBool I1 <Int 10 andBool 0 <=Int I2 andBool I2 <Int 10 
+      [simplification]
+
```

Comment:
These lemmas aren't quite true, and often it's better to just add the dis-equalities needed directly to the pre-condition of the proofs (to minimize how many assumptions we're making).

For example here: https://github.com/runtimeverification/evm-semantics/blob/master/tests/specs/mcd/end-cash-pass-rough-spec.k#L142

We add the needed disequaltiies to the pre-condition of the proof, and elsewhere we have lemmas about simplifying `#lookup(M [ K <- _ ], K') => #lookup(M, K') requires K =/=Int K' [simplificaiton]`.

Reply:
Will understand this, likely moving to test-specific files rather than `lemmas.k`.